### PR TITLE
refactor(app): remove unused amd requires

### LINKED ISSUE #2392: Investigate unused amd requires
Per @shane-tomlinson's https://github.com/mozilla/fxa-content-server/pull/2377#discussion_r29797070

> _"I don't think `p` is used anywhere in this module."_

---

I was playing w/ [**grunt-amdcheck**](https://www.npmjs.com/package/grunt-amdcheck) and noticed some interesting results ~~(but lots of noise)~~ (fixed in grunt-amdcheck@1.3.0).

Here is my _grunttasks/**amdcheck.js**_ file:

```js
module.exports = function (grunt) {
  'use strict';

  grunt.config('amdcheck', {
    app: {
      options: {
        removeUnusedDependencies: false
      },
      files: [{
        expand: true,
        cwd: 'app/',
        src: [
          '**/*.js',
          '!bower_components/**'
        ]
      }]
    }
  });
}
```

And here is the ~~difficult to read~~ output:

```
$ grunt amdcheck

Running "amdcheck:app" (amdcheck) task

app/scripts/lib/channels/fx-desktop.js (1 module)
Unused paths: lib/auth-errors

app/scripts/lib/config-loader.js (1 module)
Unused paths: underscore

app/scripts/lib/fxa-client.js (1 module)
Unused paths: underscore, lib/xhr

app/scripts/lib/oauth-client.js (1 module)
Unused paths: lib/promise, lib/session

app/scripts/models/account.js (1 module)
Unused paths: lib/auth-errors

app/scripts/models/auth_brokers/iframe.js (1 module)
Unused paths: jquery, lib/promise, lib/channels/iframe, lib/auth-errors

app/scripts/models/auth_brokers/mixins/channel.js (1 module)
Unused paths: underscore

app/scripts/models/reliers/oauth.js (1 module)
Unused paths: lib/promise

app/scripts/models/user.js (1 module)
Unused paths: lib/assertion, lib/oauth-client

app/scripts/views/complete_reset_password.js (1 module)
Unused paths: lib/validate

app/scripts/views/complete_sign_up.js (1 module)
Unused paths: lib/validate, lib/promise

app/scripts/views/confirm.js (1 module)
Unused paths: underscore

app/scripts/views/confirm_reset_password.js (1 module)
Unused paths: jquery, lib/constants

app/scripts/views/marketing_snippet_ios.js (1 module)
Unused paths: lib/constants

app/scripts/views/ready.js (1 module)
Unused paths: lib/session, lib/xss, lib/strings, lib/auth-errors, lib/promise

app/scripts/views/settings/avatar.js (1 module)
Unused paths: underscore

app/scripts/views/settings/avatar_camera.js (1 module)
Unused paths: canvasToBlob

app/scripts/views/settings/avatar_change.js (1 module)
Unused paths: underscore, lib/url

app/scripts/views/settings/avatar_crop.js (1 module)
Unused paths: underscore

app/scripts/views/sign_in.js (1 module)
Unused paths: lib/validate

app/scripts/views/sign_up.js (1 module)
Unused paths: lib/session, lib/strings

app/tests/spec/lib/app-start.js (1 module)
Unused paths: models/reliers/base

app/tests/spec/lib/assertion.js (1 module)
Unused paths: sinon, lib/constants, vendor/jwcrypto/lib/algs/rs

app/tests/spec/lib/channels/duplex.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/lib/channels/fx-desktop.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/lib/channels/iframe.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/lib/channels/null.js (1 module)
Unused paths: chai

app/tests/spec/lib/channels/web.js (1 module)
Unused paths: router, views/sign_in

app/tests/spec/lib/cropper.js (1 module)
Unused paths: jquery

app/tests/spec/lib/fxa-client.js (1 module)
Unused paths: lib/session

app/tests/spec/lib/mailcheck.js (1 module)
Unused paths: lib/promise

app/tests/spec/lib/oauth-client.js (1 module)
Unused paths: jquery, ../../lib/helpers

app/tests/spec/lib/profile-client.js (1 module)
Unused paths: jquery

app/tests/spec/lib/router.js (1 module)
Unused paths: underscore

app/tests/spec/lib/url.js (1 module)
Unused paths: underscore

app/tests/spec/lib/validate.js (1 module)
Unused paths: underscore

app/tests/spec/models/auth_brokers/base.js (1 module)
Unused paths: sinon

app/tests/spec/models/auth_brokers/fx-desktop.js (1 module)
Unused paths: underscore, lib/constants

app/tests/spec/models/auth_brokers/iframe.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/models/auth_brokers/web-channel.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/models/notifications.js (1 module)
Unused paths: lib/promise

app/tests/spec/models/profile-image.js (1 module)
Unused paths: sinon, lib/promise

app/tests/spec/models/verification/base.js (1 module)
Unused paths: sinon

app/tests/spec/models/verification/reset-password.js (1 module)
Unused paths: sinon

app/tests/spec/models/verification/sign-up.js (1 module)
Unused paths: sinon

app/tests/spec/views/change_password.js (1 module)
Unused paths: underscore

app/tests/spec/views/confirm_reset_password.js (1 module)
Unused paths: lib/session

app/tests/spec/views/cookies_disabled.js (1 module)
Unused paths: lib/promise

app/tests/spec/views/coppa/coppa-date-picker.js (1 module)
Unused paths: underscore, lib/auth-errors, lib/promise

app/tests/spec/views/force_auth.js (1 module)
Unused paths: lib/session

app/tests/spec/views/mixins/avatar-mixin.js (1 module)
Unused paths: backbone

app/tests/spec/views/mixins/back-mixin.js (1 module)
Unused paths: underscore

app/tests/spec/views/mixins/password-mixin.js (1 module)
Unused paths: backbone

app/tests/spec/views/mixins/timer-mixin.js (1 module)
Unused paths: backbone

app/tests/spec/views/oauth_sign_up.js (1 module)
Unused paths: lib/auth-errors

app/tests/spec/views/ready.js (1 module)
Unused paths: lib/promise

app/tests/spec/views/reset_password.js (1 module)
Unused paths: lib/session

app/tests/spec/views/settings.js (1 module)
Unused paths: underscore, lib/constants

app/tests/spec/views/settings/avatar.js (1 module)
Unused paths: underscore

app/tests/spec/views/settings/avatar_camera.js (1 module)
Unused paths: underscore

app/tests/spec/views/settings/avatar_change.js (1 module)
Unused paths: underscore

app/tests/spec/views/settings/avatar_crop.js (1 module)
Unused paths: underscore, draggable, lib/constants

app/tests/spec/views/settings/avatar_gravatar.js (1 module)
Unused paths: underscore

app/tests/spec/views/sign_up.js (1 module)
Unused paths: underscore, moment


Total unused dependencies: 89 in 64 files.
Total processed files: 251

Done, without errors.


Execution Time (2015-05-12 17:23:13 UTC)
loading tasks  3.7s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 56%
amdcheck:app   2.9s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 44%
Total 6.6s
```

### PR DESCRIPTION
**Pull request by riadhchtara**
Fixes #2392

### CODE REVIEW

**Comment by riadhchtara** [comment]
@vladikoff : i don't why my previous pull request was closed. Cold you please have look to the pr quickly before some new conflicts appear?
Thanks

File: app/scripts/views/settings/avatar_camera.js [hunk 1]
```diff
@@ -7,7 +7,6 @@
 define([
   'underscore',
   'cocktail',
-  'canvasToBlob',
```
Comment (vladikoff) [inline app/scripts/views/settings/avatar_camera.js]: We cannot remove this, polyfill for older browsers: https://github.com/mozilla/fxa-content-server/blob/cfe4db3b4e390725f430dc9127f7adbbb233311e/app/scripts/views/settings/avatar_camera.js#L200

Needs `exceptsPaths`

**Comment by vladikoff** [comment]
This needs to be updated: https://github.com/mozilla/fxa-content-server/pull/2448#discussion_r30837055

Secondary:

As part of a separate PR (to minimize diff) add the 
```
tests/functional/**/*.js
``` 
files to amdcheck as well.

Report for those: 
```
➜  fxa-content-server git:(riadhchtara-amd) grunt amdcheck
Running "amdcheck:app" (amdcheck) task
tests/functional/404.js (1 module)
Unused paths: intern/chai!assert

tests/functional/500.js (1 module)
Unused paths: intern/chai!assert

tests/functional/avatar.js (1 module)
Unused paths: app/scripts/lib/constants

tests/functional/bounced_email.js (1 module)
Unused paths: intern/chai!assert

tests/functional/delete_account.js (1 module)
Unused paths: intern/chai!assert

tests/functional/firefox/functional_firefox.js (1 module)
Unused paths: ./same_account

tests/functional/firefox/same_account.js (1 module)
Unused paths: intern/chai!assert

tests/functional/force_auth.js (1 module)
Unused paths: intern/node_modules/dojo/Deferred, tests/lib/restmail

tests/functional/oauth_force_email.js (1 module)
Unused paths: require

tests/functional/oauth_iframe.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/oauth_preverified_sign_up.js (1 module)
Unused paths: intern/node_modules/dojo/node!xmlhttprequest, app/bower_components/fxa-js-client/fxa-client, tests/lib/restmail

tests/functional/oauth_reset_password.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/oauth_sign_up.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/oauth_sign_up_verification_redirect.js (1 module)
Unused paths: intern, intern/node_modules/dojo/node!xmlhttprequest, tests/lib/restmail

tests/functional/oauth_sync_sign_in.js (1 module)
Unused paths: intern/node_modules/dojo/node!leadfoot/helpers/pollUntil

tests/functional/oauth_webchannel.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/oauth_webchannel_keys.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/pages.js (1 module)
Unused paths: intern/chai!assert

tests/functional/pp.js (1 module)
Unused paths: intern/chai!assert

tests/functional/settings_common.js (1 module)
Unused paths: intern/chai!assert, app/scripts/lib/constants

tests/functional/sign_in.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/sign_in_cached.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/sign_up.js (1 module)
Unused paths: intern/node_modules/dojo/node!leadfoot/helpers/pollUntil

tests/functional/sync_reset_password.js (1 module)
Unused paths: tests/lib/restmail

tests/functional/sync_settings.js (1 module)
Unused paths: intern/chai!assert, app/scripts/lib/constants

tests/functional/sync_sign_in.js (1 module)
Unused paths: intern/chai!assert, intern/node_modules/dojo/node!leadfoot/helpers/pollUntil

tests/functional/sync_sign_up.js (1 module)
Unused paths: intern/node_modules/dojo/node!leadfoot/helpers/pollUntil

tests/functional/tos.js (1 module)
Unused paths: intern/chai!assert


Total unused paths: 36 in 28 files.
Total unused dependencies: 35 in 27 files.
Total processed files: 288
Warning: Task "amdcheck:app" failed. Use --force to continue.
```

File: app/scripts/views/settings/avatar_camera.js (hunk 1 above)
Comment (riadhchtara) [inline app/scripts/views/settings/avatar_camera.js]: ah ok

**Comment by riadhchtara** [comment]
@vladikoff : thanks for the review.
How about this new patch?
I will work on the new tests/functional later, when this PR is merged as you said.

**Comment by vladikoff** [comment]
@riadhchtara Please rebase this against `master` https://github.com/mozilla/fxa-content-server/tree/master

**Comment by riadhchtara** [comment]
@vladikoff : hi ! I have just done! the rebase, could you p0lease review it quickly!
there are so many files changed here so there is a big probability that a conflict appearers.
Thanks

**Comment by pdehaan** [comment]
Great job, thanks again @riadhchtara!

**Comment by vladikoff** [comment]
Yeah this should be really useful!

**Comment by riadhchtara** [comment]
YES!!! it landed after 4 or 5 rebases.
@vladikoff @vladikoff thanks for you too guys

**Pull request merged**